src/plugins/dna_export/src/ExportSequenceTask.cpp

```diff
@@ -553,18 +553,31 @@ U2Sequence ExportAnnotationSequenceSubTask::importAnnotatedSeq2Dbi(const SharedA
     importer.startSequence(os, resultDbiRef, U2ObjectDbi::ROOT_FOLDER, ad->name, false);
     CHECK_OP(os, U2Sequence());
 
-    foreach (const U2Region &annotatedRegion, ad->location->regions) {
+    bool isReverseComplement = ad->getStrand().isCompementary() && nullptr != ei.complTT;
+    for (const U2Region &annotatedRegion : qAsConst(ad->location->regions)) {
         qint64 currentRegionLength = 0;
-        for (qint64 pos = annotatedRegion.startPos; pos < annotatedRegion.endPos(); pos += MAX_CHUNK_LENGTH) {
-            const qint64 currentChunkSize = qMin(MAX_CHUNK_LENGTH, annotatedRegion.endPos() - pos);
+        QList<QByteArray> chunkContentList;
```

Comment:
The whole idea behind MAX_CHUNK_SIZE is to avoid keeping  large memory blocks in RAM and flush these blocks into DB (importer.addBlock) independently. This way the algorithm never uses memory above MAX_CHUNK_SIZE bytes

With your impl the algorithm will require as much of memory as the annotated sequence size: up to int64.

Reply:
Yep, I get it